server/auth/auth.go

```diff
@@ -671,6 +671,27 @@ func (auth *AuthManager) Notify(acctID account.AccountID, msg *msgjson.Message,
 // Penalize signals that a user has broken a rule of community conduct, and that
 // their account should be penalized.
 func (auth *AuthManager) Penalize(user account.AccountID, rule account.Rule) error {
+	// Notify user of penalty.
+	details := "You may no longer trade. Leave your client running to complete outstanding trades."
+	penalty := &msgjson.Penalty{
+		Rule:     rule,
+		Time:     uint64(unixMsNow().Unix()),
+		Duration: uint64(time.Hour * 24 * 365 * 100), // 100 years
+		Details:  details,
+	}
+	sig, err := auth.signer.Sign(penalty.Serialize())
+	if err != nil {
+		return fmt.Errorf("signature error: %v", err)
+	}
+	penaltyNote := &msgjson.PenaltyNote{
+		Sig:     sig.Serialize(),
+		Penalty: penalty,
+	}
+	note, err := msgjson.NewNotification(msgjson.PenaltyRoute, penaltyNote)
+	if err != nil {
+		return fmt.Errorf("error creating penalty notification: %v", err)
+	}
+	auth.Notify(user, note, time.Hour*72)
```

Comment:
Any need to resend the notification on re-connect, since the `connect` response will already include the penalty?

Reply:
@JoeGruffins made the same point.  Makes sense if this goes back to a regular send if it's in the `'connect'` response.

Reply:
We can remove it later if need be.